lib/ansible/modules/packaging/os/installp.py

```diff
@@ -0,0 +1,194 @@
+    state: present
+
+# Remove packages foo and bar
+- installp:
+    name: foo,bar
+    state: absent
+'''
+
+RETURN = '''
+changed:
+    description: Return changed for installp actions as true or false.
+    returned: always
+    type: boolean
+    version_added: 2.5
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+package_actions_param = {
+    'preview': '-p',
+    'install': '',
+}
+
+accept_license_param = {
+    True: '-Y',
+    'no': '',
+}
+
+
+def remove(module, installp_cmd, packages, package_action):
+    remove_count = 0
+
+    for package in packages:
+
+        rc, remove_out, err = module.run_command('%s -u %s %s' % (
+            installp_cmd,
+            package,
+            package_actions_param[package_action]))
+        if rc != 0:
+            module.fail_json(msg="Failed to uninstall the package %s" %
+                                 package, rc=rc, err=err)
+
+        else:
+            remove_count += 1
+
+    if remove_count > 0:
+        module.exit_json(changed=True, msg="%s packages removed" %
+                                           remove_count)
+    else:
+        module.exit_json(changed=False, msg="No packages removed")
+
+
+def install(module, installp_cmd, packages, repository_path, package_action,
+            accept_license):
+    install_count = 0
+
+    for package in packages:
+
+        rc, install_out, err = module.run_command('%s -a %s -X %s -d %s %s'
+                                                  % (installp_cmd,
+                                                     package_actions_param[
+                                                         package_action],
+                                                     accept_license_param[
+                                                         accept_license],
+                                                     repository_path,
+                                                     package
+                                                     ))
+        if rc != 0:
+            module.fail_json(msg="Failed to install the package %s" % package,
+                             rc=rc, err=err)
+        else:
+            install_count += 1
+
+    if install_count > 0:
+        module.exit_json(changed=True, msg="%s packages installed" %
+                                           install_count)
+    else:
+        module.exit_json(changed=False, msg="No packages installed")
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(aliases=["pkg"], required=True),
+            repository_path=dict(type='str', default=None),
+            accept_license=dict(type='bool', default='no'),
+            state=dict(default="present", choices=["present", "absent"]),
+            package_action=dict(default="preview",
+                                choices=["preview", "install"]),
+        ))
+
+    installp_params = module.params
+    pkgs = installp_params["name"].split(",")
```

Comment:
Rather than manually split here, you should define this as a list in the argument_spec:

``` python
            name=dict(aliases=["pkg"], required=True, type='list'),
```

An argument_spec list can either be handed a yaml list or it can receive a comma separated string (which it will automatically split into a list)